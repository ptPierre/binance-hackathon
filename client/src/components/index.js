export { default as Header } from './header/Header';
export { default as HeaderControls } from './header-controls/HeaderControls';
export { default as HeaderLogo } from './header-logo/HeaderLogo';
export { default as HeaderNav } from './header-nav/HeaderNav';
export { default as ErrorBoundary } from './error-boundary/ErrorBoundary';
