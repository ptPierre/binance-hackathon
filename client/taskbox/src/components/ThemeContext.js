import React from 'react';

const ThemeContext = React.createContext("#eee");

export default ThemeContext;
