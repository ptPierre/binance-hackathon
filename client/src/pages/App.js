import React from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import Header from '../components/header/Header';
import { ErrorBoundary } from '../components';

function App() {
  return (
    <ErrorBoundary>
      <CssBaseline />
      <Header />
    </ErrorBoundary>
  );
}

export default App;
