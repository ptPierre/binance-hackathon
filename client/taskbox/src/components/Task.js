import React, { useContext } from 'react';
import { func, shape, string } from 'prop-types';
import ThemeContext from './ThemeContext';

export default function Task({
  task: { id, title, state },
  onArchiveTask,
  onPinTask,
}) {
  const color = useContext(ThemeContext);

  return (
    <div className={`list-item ${state}`}>
      <label className="checkbox">
        <input
          type="text"
          defaultChecked={state === 'TASK_ARCHIVED'}
          name="checked"
          disabled={true}
        />
        <span className="checkbox-custom" onClick={() => onArchiveTask(id)} />
      </label>
      <div className="title">
        <input
          type="text"
          value={title}
          readOnly={true}
          placeholder="Input title"
        />
      </div>

      <div className="actions" onClick={(event) => event.stopPropagation()}>
        {state !== 'TASK_ARCHIVED' && (
          <button
            style={{ backgroundColor: 'inherit' }}
            onClick={() => onPinTask(id)}
          >
            <span
              style={{ color: state === 'TASK_INBOX' ? color : '' }}
              className="icon-star"
            />
          </button>
        )}
      </div>
    </div>
  );
};

Task.propTypes = {
  task: shape({
    /** ID of the task */
    id: string.isRequired,
    /** Title of the task */
    title: string.isRequired,
    /** Current state of the task */
    state: string.isRequired
  }),
  /** Event to change the task to archived */
  onArchiveTask: func,
  /** Event to change the task to pinned */
  onPinTask: func
};
