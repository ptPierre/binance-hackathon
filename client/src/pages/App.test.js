import { render } from '@testing-library/react';
import { Header, HeaderControls, HeaderNav, HeaderLogo } from '../components';
import App from './App';

const Components = [
  App,
  Header,
  HeaderControls,
  HeaderLogo,
  HeaderNav
];

const props = {
  HeaderLogo: {
    text: 'Test',
  },
  HeaderNav: {
    navItems: [
      { text: 'Test1' },
      { text: 'Test2' },
    ],
  },
};

describe('Component Snapshots', () => {
  Components.forEach((Component) => {
    test(`${Component.name}`, () => {
      const { asFragment } = render(<Component {...props[Component.name]} />);
      const appRender = asFragment();
      expect(appRender).toMatchSnapshot();
    });
  });
});
