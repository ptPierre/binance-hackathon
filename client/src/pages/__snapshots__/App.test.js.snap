// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Snapshots App 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolBar-1 MuiToolbar-gutters"
    >
      <div
        class="makeStyles-root-2"
        style="margin-right: 3rem;"
      >
        <svg
          class="makeStyles-logo-3"
        >
          ninja-head.svg
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1"
          style="font-size: 16px; align-self: center;"
        >
          NinjaSwap
        </p>
      </div>
      <div
        class="makeStyles-root-4"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Swap
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Pool
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Stake
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Vote
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-wallet-5 MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Connect a Wallet
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Component Snapshots Header 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolBar-6 MuiToolbar-gutters"
    >
      <div
        class="makeStyles-root-7"
        style="margin-right: 3rem;"
      >
        <svg
          class="makeStyles-logo-8"
        >
          ninja-head.svg
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1"
          style="font-size: 16px; align-self: center;"
        >
          NinjaSwap
        </p>
      </div>
      <div
        class="makeStyles-root-9"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Swap
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Pool
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Stake
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Vote
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-wallet-10 MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Connect a Wallet
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`Component Snapshots HeaderControls 1`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-wallet-11 MuiButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Connect a Wallet
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33C4.62 15.49 4 13.82 4 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
            />
          </svg>
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Component Snapshots HeaderLogo 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-12"
    style="margin-right: 3rem;"
  >
    <svg
      class="makeStyles-logo-13"
    >
      ninja-head.svg
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body1"
      style="font-size: 16px; align-self: center;"
    >
      Test
    </p>
  </div>
</DocumentFragment>
`;

exports[`Component Snapshots HeaderNav 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-14"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Test1
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Test2
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
